python/cuml/dask/decomposition/base.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.dask.common import raise_exception_from_futures
+from cuml.dask.common.comms import worker_state, CommsContext
+
+from cuml.dask.common.input_utils import to_output
+
+from cuml.dask.common.part_utils import flatten_grouped_results
+
+from dask.distributed import wait
+
+from uuid import uuid1
+
+from cuml.dask.common.base import BaseEstimator
+from cuml.dask.common.input_utils import DistributedDataHandler
+
+
+class BaseDecompositionFitMixin(BaseEstimator):
```

Comment:
Is this a mixin? I think it's a base class instead

Reply:
It is a base class but it's also a mixin because it's providing the `_fit()` function for the subclasses to call. I consolidated the common constructor logic here just so it didn't need to be duplicated across PCA/TSVD. I felt adding both `Base` and `Mixin` to the names was appropriate since it's a base class but also a mixing. I'll split the the constructor out into a separate class, something like `BaseDecompositionEstimator` and have PCA / TSVD extend both the mixin and the base class. Sound good?